src/toxcore/dht/daemon_state.rs

```diff
@@ -113,6 +145,11 @@ mod tests {
         let serialized_len = serialized_vec.len();
         assert!(DaemonState::deserialize_old(&alice, &serialized_vec[..serialized_len - 1]).wait().is_err());
 
+        // test with serialized data corrupted
+        let mut serialized_vec = DaemonState::serialize_old(&alice);
+        serialized_vec.reverse();
```

Comment:
Why not just define an array? It seems a bit strange that it's reversed to make it invalid.

Reply:
done